Lib/test/test_unittest/testmock/testmock.py

```diff
@@ -1556,6 +1556,15 @@ def f(x=None): pass
             mock.assert_has_calls([call()])
         self.assertIsNone(cm.exception.__cause__)
 
+        uncalled_mock = Mock()
+        with self.assertRaisesRegex(
+                AssertionError,
+                '^{}$'.format(
+                    re.escape('Calls not found.\n'
+                              'Expected: [call()]\n'
+                              '  Actual: []'))) as cm:
+            uncalled_mock.assert_has_calls([call()])
+        self.assertIsNone(cm.exception.__cause__)
```

Comment:
The same goes with this assertion. Why is it important that `__cause__` is None? 
I'd suggest factoring out a `self.assertCauseIsNone(cm.exception)` and then a docstring or comment in that method explaining why it's important.

Reply:
I don't think it's necessarily important that `__cause__` is `None` here, it just has no meaning to be anything other than `None`, but the test is here to contrast with below where it should be a `TypeError`.